tokio/src/runtime/time/entry.rs

```diff
@@ -490,8 +498,10 @@ impl TimerEntry {
     fn inner(&self) -> &TimerShared {
         let inner = unsafe { &*self.inner.get() };
         if inner.is_none() {
+            let shard_id =
+                super::rand::thread_rng_n(self.driver.driver().time().inner.get_shard_size());
```

Comment:
I'm wondering whether we could use the current worker id, when this is called from a runtime thread. What do you think?

Reply:
It's worth a try, this is the difference between `sleep`s and `task`s. The timer wheel is more like an mpsc, while the OwnedTasks I optimized before are mpmc. That is to say, when it comes to lock contention in sleep, the most severe one is insert, while clear is much smaller in lock contention. Therefore, making the create operation thread-local is worth trying for `sleep`s.

However, I still need to point out that it would be useful if our scenario involves creating sleep and timeouts in a worker thread environment. But when it needs to be created in a thread outside of the worker thread, the effect is minimal. If there are enough shards, then the performance consumed by locking is already small enough, such as the `0.79%` in the above benchmark.

Reply:
I added this logic, as expected, creating a sleep in the worker thread is almost lock free contention.

Reply:
> I added the thread-local logic, as expected, creating a sleep in the worker thread is almost lock free contention.

Assume the same would be said for creating a *timeout* in the worker thread? That it would be an almost lock free operation?

Reply:
Yes, it is almost lock-free.